docs/development/CodingStyle.md

```diff
@@ -0,0 +1,259 @@
+
+In the .h file:
+```
+#ifdef MODULENAME_INTERNALS_
+… declarations …
+#endif
+```
+In the module .c file, and in the test file but nowhere else, put `#define MODULENAME_INTERNALS_` just before including the .h file.
+
+Note: You can get the same effect by putting the internals in a separate .h file.
+
+##Implementation
+Keep functions short and distinctive.
+Think about unit test when you define your functions. Ideally you should implement the test cases before implementing the function.
+
+Never put multiple statements on a single line.
+Never put multiple assignments on a single line.
+Never put multiple assignments in a single statement.
+
+Defining constants using pre-processor macros is not preferred.
+Const-correctness should be enforced.
+This allows some errors to be picked up at compile time (for example getting the order of the parameters wrong in a call to memcpy).
+
+A function should only read data from the HW once in each call, and preferably all at one place.
+For example, if gyro angle or time is needed multiple times, read once and store in a local variable.
+
+Use `for` loops (rather than `do` or `while` loops) for iteration.
+
+The use of `continue` or `goto` should be avoided.
+Same for multiple `return` from a function and multiple `break` inside a `case`.
+In general, they reduce readability and maintainability.
+In rare cases such constructs can be justified but only when you have considered and understood the alternatives and still have a strong reason.
+
+Use parentheses around each group in logical and mathematical statements,
+rather than relying on the implicit logic and operator priority.
+The compiler knows what it’s doing but it should be easy for people too.
+
+#Includes
+All files must include their own dependencies and not rely on includes from the included files or that some other file was included first.
+
+Do not include things you are not using.
+
+"[#pragma once](https://en.wikipedia.org/wiki/Pragma_once)" is preferred over "#include guards" to avoid multiple includes.
+
+
+#Other details
+No trailing whitespace at the end of lines or at blank lines.
+
+Stay within 120 columns, unless exceeding 120 columns significantly increases readability and does not hide information.
+(Less is acceptable. More than 140 makes it difficult to read on Github so that should be a hard limit.)
+
+Take maximum possible advantage of compile time checking, so generally warnings should be as strict as possible.
+
+Don't call or reference "upwards". That is don't call or use anything in a software layer that is above the current layer. The software layers are not that obvious in Cleanflight, but we can certainly say that device drivers are the bottom layer and so should not call or use anything outside the device drivers.
+
+Target specific code (e.g. #ifdef CC3D) should be absolutely minimised.
+
+typedef void handlerFunc(void); is easier to read than typedef void (*handlerFuncPtr)(void);.
```

Comment:
this should be 
`typedef void handlerFunc(void);` is easier to read than `typedef void (*handlerFuncPtr)(void);`.